eks_addon/release.sh

```diff
@@ -0,0 +1,174 @@
+
+  # Helm login - set -e will fail on error
+  echo "$AUTH_AWS_ECR_PASSWORD" | helm registry login --username AWS --password-stdin "$ECR_REGISTRY"
+
+  echo "✅ AWS Authentication successful"
+}
+
+copy_docker_image_to_ecr() {
+  # Get the otelcol repository from values.yaml
+  local src_repo=$(yq e ".image.otelcol.repository" "${CHART_DIR}/values.yaml")
+  [[ -n "$src_repo" && "$src_repo" != "null" ]] || { echo "❌ Error: Could not find otelcol repository in values.yaml"; exit 1; }
+
+  local src="${src_repo}:${CHART_APPVERSION}"
+  local dst="${ECR_OTELCOL_REPO}:${CHART_APPVERSION}"
+
+  echo "⏳ Copying otelcol image from ${src} to ${dst} ..."
+  docker pull ${src}
+  docker tag ${src} ${dst}
+  ${DRY_RUN_PREFIX} docker push ${dst}
+  echo "✅ Successfully copied otelcol image: ${src} → ${dst}"
+}
+
+modify_helm_chart() {
+  local chart_dir="$1"
+  local overrides_dir="${SCRIPT_DIR}/overrides"
+
+  echo "⏳ Removing subcharts ..."
+  rm -rf "${chart_dir}/charts"
+
+  echo "⏳ Modifying Helm chart in ${chart_dir} using overrides from ${overrides_dir} ..."
+  for override in "${overrides_dir}"/*.yaml; do
+    if [[ -f "${override}" ]]; then
+      local override_basename=$(basename "${override}")
+      local target_file="${chart_dir}/${override_basename}"
+
+      # Process environment variables
+      local tmp_file="/tmp/${override_basename}.expanded"
+      eval "cat <<EOF
+$(cat "${override}")
+EOF" > "${tmp_file}"
+
+      # Merge override with the corresponding file in the chart
+      yq eval-all 'select(fileIndex==0) * select(fileIndex==1)' -i "${target_file}" "${tmp_file}"
+      rm -f "${tmp_file}"
+    fi
+  done
+
+  echo "⏳ Moving values.schema.json to aws_mp_configuration_schema.json and removing unsupported properties ..."
+  cp "${tmp_chart_dir}/values.schema.json" "${tmp_chart_dir}/aws_mp_configuration_schema.json"
+  disabled_properties=(
+    "enabled"
+    "operatorcrds"
+    "operator-crds"
+    "operator"
+    "opentelemetry-operator"
+    "instrumentation"
+    "certmanager"
+    "cert-manager"
+    "targetAllocator"
+  )
+  for prop in "${disabled_properties[@]}"; do
+      yq e "del(.properties.\"${prop}\")" -i "${tmp_chart_dir}/aws_mp_configuration_schema.json"
+  done
+
+  echo "✅ Successfully modified the Helm chart for EKS Add-on compliance"
+}
+
+package_and_push_helm_chart() {
+  local ecr_chart_release="oci://${ECR_HELM_REPO}:${CHART_VERSION}"
+
+  # Check if chart already exists in the registry
+  if helm show chart "${ecr_chart_release}" &>/dev/null; then
+    echo "❌ Chart already exists in registry: ${ecr_chart_release}. Use --chart-version to push another version of this chart."
+    return 1
+  fi
+
+  # Copy chart to a temporary build directory
+  local build_dir="${SCRIPT_DIR}/build"
+  rm -rf "${build_dir}"
+  mkdir -p "${build_dir}"
+  cp -R "${CHART_DIR}" "${build_dir}/"
+
+  # Function to modify the Helm chart to meet EKS Add-on requirements
```

Comment:
It might be better to move this comment to the method itself, and make it a bit more descriptive of why the modifications are needed.